### ISSUE TITLE
Code Table Request - identifiers - ARK

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#6443]
## Initial Request

ARK [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=CTCOLL_OTHER_ID_TYPE#ark) including base URL --> identifier

Data:

[temp_ark.csv.zip](https://github.com/ArctosDB/arctos/files/11837578/temp_ark.csv.zip)

Summary:

```

 guid_prefix | numrecs 
-------------+---------
 DMNS:Mamm   |       4
 MSB:Host    |       4
 MSB:Para    |       1
 UAMObs:Ento |       1
 UCM:Fish    |      70
 UCM:Herp    |     411
 UCM:Mamm    |       1
```

 @ebraker
 @msbparasites
 @mlbowser
 @campmlc
 @DerekSikes
 @acdoll
 @jrdemboski

@ArctosDB/arctos-code-table-administrators 

## Approval

_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [ ] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [ ] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation

_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

_Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel._

_Make changes as described above. Ensure the URL of this Issue is included in the definition._

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by mkoo** [comment]
ARK is a kind of DOI and like DOI more than one entity can mint it. (for example Geome-db for the amphibian disease portal) so I dont think converting to ```identifier``` is necessarily the way to go. It could be converted to doi as it is technically and logistically a kind of DOI but then the prepend is different.

So I think this IDtype of ```ARK``` should stay with the prepend of "http://n2t.net/ark:/" but allow Issuedby to be the Smithsonian or AmphibianDisease.org or whomever. Makes sense?

And no, I have no identifiers for the Amphibian Disease portal but just you wait!

**Comment by dustymc** [comment]
>Makes sense?

No. Like DOI "ARKness" isn't informative - as you say, anyone can get one (or a billion) and use them for anything. There's no *possible* knowledge in the type.

See also attached data, complexity demonstrably leads to broken identifiers, simplicity is bliss. I can see no reason not to simplify.

**Comment by campmlc** [comment]
I truly don't understand the sense of urgency in compelling users to convert all ID types to "identifier".I might want to specify a search for an ark id issues by the Smithsonian and exclude institutional catalog numbers from the same agent. Merging both those ID types into " identifier" makes that impossible. I can see the need to reduce specific IDs in the code table. But this rapid abandonment of potential information without prior testing and assessment seems rash.

**Comment by dustymc** [comment]
>abandonment of potential information

There is no possible way that this can be the case.

The current data are "goes through a redirection service."

The proposal is precisely the same thing, with less needless complexity.

This proposal has significant functional implications regarding the ESN.

If I get hit by a meteorite after the delete and before the insert (which isn't how this happens, so that seems pretty unlikely!) or a bunch of us have somehow simultaneously misunderstood the model or something, there's CSV just above. This cannot possibly be lossy.

**Comment by campmlc** [comment]
I just want everyone to be aware that if this goes through, it would make
it extremely difficult for collection managers to search for ARK IDs or
DOIs associated with their collections. You would have to search for all
"identifiers" issued by each individual agent. If no one cares, so be it.
But from a collection managers perspective, this is losing data
discoverability.

**Comment by mkoo** [comment]
I think we're talking past each other again (a danger of forums!)
I think I see the need to retain a few of our IDtypes-- in this issue ```ARK```. Agreed there is nothing ARKness that needs to be it's own special category except that an entity could (and do ) mint both doi and ark. So let's recognize that! (I mean, you could still search of arks even it is no longer a IDtype ...)

Here's an example from https://arctos.database.museum/guid/MSB:Host:1239 where Teresa added some relationships to illustrate the differences between identifiers and relationships
![Firefox_Screenshot_2023-06-23T02-00-56 502Z](https://github.com/ArctosDB/arctos/assets/2523089/f07001b4-44c0-4ef5-b01e-209c75f8d410)

In this case we are talking about idtype=ARK -- the last row of the Relationship table is what is currently the case. (I thought we were making progress today !) We are proposing that the top row  (what Teresa added) is the new improved case. Thus, retain a generic IDTYPE of Ark but explicitly indicate the Issued by (here, the Smithsonian, Mammals Div). 
![Firefox_Screenshot_2023-06-23T02-01-12 663Z](https://github.com/ArctosDB/arctos/assets/2523089/6f6ef02a-66c6-40f3-9300-25e476086860) Clearly NMNH also issues other unique numbers like the catalog number there. 

These are all arguably Identifiers (not Relationships but let's not add another dimension of merging all of that too!). It will be much clearer to users where these come from eventually

Once we have consensus, then Dusty can do _something_ (not extra work for the collections)

(ps. I see a little typo in the value, please remove the prefixed "/" just start with the number)

**Comment by campmlc** [comment]
Yes, thanks @mkoo  - this is what I've been trying to communicate. I absolutely agree with the need for adding the issued by agent. But we need to keep ARK as a generic ID type that can be "issued by" different agents. So everything @Jegelewicz did to https://arctos.database.museum/guid/MSB:Host:1239 is perfectly OK with me. What I disagree with is for all ARK identifiers to be converted to ID type = identifier, which is how this issue was originally posted.

**Comment by dustymc** [comment]
>talking past each other again (a danger of forums!)

Yes, somehow - hence the offer/plea to zoom, wherever it went. SOMEONE (everyone?) clearly isn't understanding SOMETHING, let's please don't let this devolve into whatever it was a while back, whatever that takes. (I'd offer to meet halfway over a beer, but I think that's Needles...)

>illustrate the differences between identifiers and relationships

That should be a topic of our chat: They are literally the same thing.

>see a little typo 

That's not a "little typo," that is a symptom of the "kill the Extended Specimen Network"-level problem I'm trying to address here! That is obviously and demonstrably (scroll up and look at the attached CSV!) inevitable where we are, and damned near impossible from where I'm trying to get us! 

>make it extremely difficult for collection managers to search for ARK IDs or DOIs associated with their collections. 

How can I convince you that that's not the case? I can do this in test, I can find some other data and we can pretend it's an ARK, if you'll give the OK I can temporarily duplicate some data on some of your records, what's it take?  I am willing to demonstrate in whatever way you will accept how any of these alleged obstacles are not actually obstacles, and I think @Jegelewicz would be willing to do the same if for some reason that's better.

**Comment by mkoo** [comment]
I did say zoom tomorrow, right?! what's a good time before 10 PDT?
I suspect this will be just fine but maybe I'm not explaining my (our) selves well so midway meeting of the minds is needed! (Beer later at Needles!)

**Comment by dustymc** [comment]
I can make any time.

**Comment by Jegelewicz** [comment]
> I see a little typo in the value, please remove the prefixed "/" just start with the number)

The prepend needs to go the way of the Dodo!

I can also make any time

**Comment by dustymc** [comment]
>I added another identifier:

I notice you didn't pick the "wrong" place to split it too!

>some of the actual ARKs aren't typed as ARK! 

Yes, this is a recurring theme, grab most any of these issues and dig around in the data a bit, you'll find that. Types that don't DO STUFF (like, get you to a tube in your local freezer) - and all URIs are the same from that perspective - just (demonstrably) serve to confuse people.

>you can absolutely find ALL ARKs by searching for the part of an ARK 

yes

>http://n2t.net/ark:/

There's some technical "well actchually...." around that, but yes, all ARKs have an ARK resolver.

>going to know that

Much of the point of ARK and DOI is that nobody needs to know that - they just work - except maybe some CI/CM for some accounting purposes, and "we" are always around to help with that. (And so I don't understand any aspect of this latest barrier.)

**Comment by mkoo** [comment]
Referencing #6442 (sorry I keep flipping between these two)-- After call wtih Mariel, she's cleaned up the two DOI and ARK records manually. @ebraker has the most ARK's right now: Are these are all Morphosource links? I think Dusty needs to know who to issue it to. Otherwise it will have to be done manually
(ps. I created MorphoSource: https://arctos.database.museum/agents.cfm?agent_id=21348945)

The problem with ARK is that it's not easily intuited who issued it... another reason to have it explicit in Issuer field!

**Comment by dustymc** [comment]
>know who to issue it to. Otherwise it will have to be done manually

I can, but simply doing what's been proposed maintains the level of information that exists now, and any sidetracks that add any time to this, particularly when there's a dedicated loader that can be used to add issuer at any time, seem not-great. Can we just proceed with this (and 6442)?

(And I should be able to pull some level of metadata for ARKs and DOIs, which I'll be happy to do after we're not stuck here!)

**Comment by Jegelewicz** [comment]
We have a tool to add issued by! - https://arctos.database.museum/loaders/bulk_identifier_issuedby.cfm

**Comment by dustymc** [comment]
>tool

Yep, that's the one. Feel free to check a box while you're here....

**Comment by mkoo** [comment]
I think this can proceed but there's a few things to note so you have an Issuer

Taking a look at the csv:
all the arks that start with **65665/** are from the Smithsonian (mammal division or fish division)
all the arks starting with **87602/** are MorphoSource (I made them an agent profile already)

I checked a box already (I think) so @Jegelewicz ?

**Comment by Jegelewicz** [comment]
Emily checked the first box. I checked the second.
